### PR TITLE
Tempus: Improved testing; added testing of xDot. #2716

### PR DESCRIPTION
**Pull request by ccober6**
* Added to the tests, the testing of xDot.  Previously no testing of
   the time derivative was done.
 * Cleaned up testing to use some common functions for writing the
   solution, reference solution, and order/error convergence data.
 * Fix many miss-named and miss-spelled variables and filenames.
 * Added Irina's #defines to all tests for each Teuchos unit test.
   This makes it easy to turn off unit tests while debugging.

Others
 * Change default SolutionHistory::StorageType from "Unlimited" to
 * "Undo".
   this better reflects the normal situation for applications.  For most
   Steppers, two SolutionStates are required for implicit solvers and
   setting the time derivative.
 * Added exceptions to Steppers to ensure that there is the minimum
   number of SolutionStates available in the SolutionHistory.  Most
   Steppers require at least two (the currentState, x_{n-1}, and the
   workingState, x_n).
 * BUG FIX - In SolutionHistory::initWorkingState(), the last State is
   removed and then the CurrentState is copied into the newState.
   However when the number of States was one, there is no State to copy
   from.
 * Added doxygen comments on TimeStepControlStrategy_BasicVS, describing
   the functionality, mathematical statements, and suggested settings.
   Changed the default eta's to reflect constant time step control.
   Made these changes also in the tests.
 * Separate the "Constant" and "Variable" output control in
   TimeStepControl::getNextTime().
 * Set min and max timestep to the initial time step for constant time
   step, and added an initialize() to reset values in TimeStepControl.

@trilinos/tempus 

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ x] My code follows the code style of the affected package(s).
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ x] I have added tests to cover my changes.
- [ x] All new and existing tests passed.
- [ x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

File: packages/tempus/test/DIRK/Tempus_DIRKTest.cpp [hunk 1]
```diff
@@ -322,88 +335,90 @@ TEUCHOS_UNIT_TEST(DIRK, SinCos)
...
+        StepSize.push_back(0.0);
+        auto solution = Thyra::createMember(model->get_x_space());
+        Thyra::copy(*(model->getExactSolution(time).get_x()),solution.ptr());
+        solutions.push_back(solution);
+        auto solutionDot = Thyra::createMember(model->get_x_space());
+        Thyra::copy(*(model->getExactSolution(time).get_x_dot()),
+                    solutionDot.ptr());
+        solutionsDot.push_back(solutionDot);
+      }
+    //Teuchos::TimeMonitor::summarize();
```
Comment (sconde) [inline packages/tempus/test/DIRK/Tempus_DIRKTest.cpp]: Did you mean to comment this out?

File: packages/tempus/test/DIRK/Tempus_DIRKTest.cpp [hunk 2]
```diff
@@ -322,88 +335,90 @@ TEUCHOS_UNIT_TEST(DIRK, SinCos)
...
+    RCP<Tempus::Stepper<double> > stepper = integrator->getStepper();
+    double order = stepper->getOrder();
+    writeOrderError("Tempus_"+RKMethod_+"_SinCos-Error.dat",
+                    stepper, StepSize,
+                    solutions,    xErrorNorm,    xSlope); //,
+                    //solutionsDot, xDotErrorNorm, xDotSlope);
+
+    TEST_FLOATING_EQUALITY( xSlope,               order, 0.01   );
+    TEST_FLOATING_EQUALITY( xErrorNorm[0], RKMethodErrors[m], 5.0e-4 );
+    //TEST_FLOATING_EQUALITY( xDotSlope,            order, 0.01   );
```
Comment (sconde) [inline packages/tempus/test/DIRK/Tempus_DIRKTest.cpp]: no `xDot` tests?

File: packages/tempus/src/Tempus_TimeStepControl_impl.hpp [hunk 3]
```diff
@@ -58,8 +58,8 @@ void TimeStepControl<Scalar>::getNextTimeStep(
     Teuchos::OSTab ostab(out,1,"getNextTimeStep");
     bool printChanges = solutionHistory->getVerbLevel() !=
                         Teuchos::as<int>(Teuchos::VERB_NONE);
-    bool printChangesHi = solutionHistory->getVerbLevel() >=
-                          Teuchos::as<int>(Teuchos::VERB_HIGH);
+    //bool printChangesHi = solutionHistory->getVerbLevel() >=
```
[the remark this answers is not in the document]
Comment (ccober6) [reply packages/tempus/src/Tempus_TimeStepControl_impl.hpp]: No, not really.  We can remove it.

File: packages/tempus/test/DIRK/Tempus_DIRKTest.cpp (hunk 2 above)
Comment (ccober6) [reply packages/tempus/test/DIRK/Tempus_DIRKTest.cpp]: Yeah, currently our DIRK algorithms do not calculate xDot.  I know you have means to compute these.  How hard would it be to add the xDot at the workingState in DIRK::takeStep?

**Comment by ccober6** [comment]
@sconde Just committed suggested changes.  Do you want to re-review?

**Comment by ccober6** [comment]
@rppawlo, it looks like @sconde may be off today.  He approved it yesterday with some suggestions.  I just committed those changes.  Can you do a quick review of this, so it gets merged before my travel next week?

**Comment by rppawlo** [comment]
Working on it now

**Pull request merged**